File: plot/src/main/java/net/fwitz/math/plot/binary/split/SplitComplexFunctionRenderer.java
package net.fwitz.math.plot.binary.split;

import net.fwitz.math.complex.SplitComplex;
import net.fwitz.math.plot.binary.BinaryNumberFunctionPanel;
import net.fwitz.math.plot.binary.BinaryNumberFunctionRenderer;
import net.fwitz.math.plot.renderer.filter.ValuesFilters;

import static net.fwitz.math.complex.SplitComplex.splitComplex;

public class SplitComplexFunctionRenderer<V> extends BinaryNumberFunctionRenderer<SplitComplex, V> {
    public SplitComplexFunctionRenderer(BinaryNumberFunctionPanel<SplitComplex, V> panel, Class<V> valueType,
                                        int width, int height,
                                        ValuesFilters<V> valuesFilters) {
        super(panel, valueType, width, height, valuesFilters);
    }

    @Override
    protected SplitComplex value(double x, double y) {
        return splitComplex(x, y);
    }
}

File: plot/src/main/kotlin/net/fwitz/math/plot/binary/split/SplitComplexFunctionRenderer.kt
package net.fwitz.math.plot.binary.split

import net.fwitz.math.binary.split.SplitComplex.Companion.splitComplex
import net.fwitz.math.plot.binary.BinaryNumberFunctionPanel
import net.fwitz.math.plot.renderer.filter.ValuesFilters
import net.fwitz.math.plot.binary.BinaryNumberFunctionRenderer
import net.fwitz.math.binary.split.SplitComplex

class SplitComplexFunctionRenderer<V>(
    panel: BinaryNumberFunctionPanel<SplitComplex, V>, valueType: Class<V>,
    width: Int, height: Int,
    valuesFilters: ValuesFilters<V>
) : BinaryNumberFunctionRenderer<SplitComplex, V>(panel, valueType, width, height, valuesFilters) {
    override fun value(x: Double, y: Double): SplitComplex {
        return splitComplex(x, y)
    }
}